Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -105,10 +183,46 @@
         return m_isFigureOpen;
     }
 
+    STDMETHOD_(bool, IsGeometryStarted)() {
+        return m_hasGeometryStarted;
+    }
+
+    STDMETHOD_(D2D1_POINT_2F, GetCurrentPoint)() {
+        return m_lastPoint;
+    }
+
+    STDMETHOD_(void, SetCurrentPoint)(D2D1_POINT_2F currentPoint) {
+        m_lastPoint = currentPoint;
+    }
+
+    STDMETHOD_(D2D1_POINT_2F, GetStartingPoint)() {
+        return m_startPoint;
+    }
+
+    STDMETHOD_(void, SetStartingPoint)(D2D1_POINT_2F startPoint) {
+        m_startPoint = startPoint;
+    }
+
+    STDMETHOD_(void, SetBackingSink)(ID2D1GeometrySink* sink) {
+        m_geometrySink = sink;
+    }
+
+    STDMETHOD_(void, AllowAllExceptFinalEndFigure)(bool ignore) {
```

Comment:
>ignore [](start = 56, length = 6)

From the name of the method, and the variable you are setting, it looks like m_allowsFinalFigureCall = !ignore?  Also, will pick a different name for the variable.